packages/form-js-viewer/src/render/components/form-fields/parts/Datepicker.js

```diff
@@ -0,0 +1,183 @@
+        instance.close();
+      }
+    };
+
+    // Remove dirty tag to have mouse day selection prioritize input blur
+    const onCalendarMouseDown = (e) => {
+      if (e.target.classList.contains('flatpickr-day')) {
+        setIsInputDirty(false);
+      }
+    };
+
+    // TODO: MORE DETAIL FOR COMMENTS
+    instance.config.onOpen = [
+      () => instance.calendarContainer.addEventListener('focusout', onCalendarFocusOut),
+      () => instance.calendarContainer.addEventListener('mousedown', onCalendarMouseDown),
+    ];
+
+    instance.config.onClose = [
+      () => instance.calendarContainer.removeEventListener('focusout', onCalendarFocusOut),
+      () => instance.calendarContainer.removeEventListener('mousedown', onCalendarMouseDown),
+      () => { setForceRefreshValue(true); }
+    ];
+
+  }, [ disabled, disallowPassedDates, enterKeyboardEvent ]);
+
+  // Shorts the date value back to the source
+  useEffect(() => {
+
+    if (!flatpickrInstance || !flatpickrInstance.config) return;
+
+    flatpickrInstance.setDate(date, true);
+
+  }, [ flatpickrInstance, date.toString() ]);
+
+  // OnChange is updated dynamically, so not to re-render the flatpicker every time it changes
+  useEffect(() => {
+
+    if (!flatpickrInstance || !flatpickrInstance.config) return;
+
+    flatpickrInstance.config.onChange = [ (date) => setDate(new Date(date)) ];
+
+  }, [ flatpickrInstance, setDate ]);
+
+  const focusRelevantCalendarDay = useCallback(
+    (e) => {
+
+      if (!flatpickrInstance) return;
+
+      const container = flatpickrInstance.calendarContainer;
+      const dayToFocus =
+        container.querySelector('.flatpickr-day.selected') ||
+        container.querySelector('.flatpickr-day.today') ||
+        container.querySelector('.flatpickr-day');
+
+      // @ts-ignore
+      dayToFocus && dayToFocus.focus();
+    },
+    [ flatpickrInstance ]
+  );
+
+  const onInputKeyDown = useCallback(
+    (e) => {
+
+      if (!flatpickrInstance) return;
+
+      if (e.code === 'ArrowDown') {
+        !flatpickrInstance.isOpen && flatpickrInstance.open();
+        focusRelevantCalendarDay();
+        e.preventDefault();
+      }
+    },
+    [ flatpickrInstance, focusRelevantCalendarDay ],
+  );
+
+  const onInputFocus = useCallback(
+    (e) => {
+
+      if (!flatpickrInstance || focusScopeRef.current.contains(e.relatedTarget)) return;
+
+      flatpickrInstance.open();
+    }, [ flatpickrInstance ]
+  );
+
+  // Simulate an enter press on blur to make sure the date value is submitted in all scenarios
+  const onInputBlur = useCallback(
+    (e) => {
+
+      if (!isInputDirty || !e.relatedTarget || e.relatedTarget.classList.contains('flatpickr-day')) return;
+      setForceRefreshValue(true);
+
+    }, [ isInputDirty ]
+  );
+
+  return <InputAdorner
+    pre={ <CalendarIcon /> }
+    disabled={ disabled }
+    rootRef={ focusScopeRef }
+    inputRef={ dateInputRef }>
+    <div style={ { width: '100%' } }>
```

Comment:
Should we introduce another selector for this wrapper?

`fjs-datepicker`

Reply:
Yup, done 👍